File: Problem Solving/Algorithms/Implementation/Apple and Orange.c
void countApplesAndOranges(int s, int t, int a, int b, int apples_count, int* apples, int oranges_count, int* oranges)
{
    int landedApples = 0, landedOranges = 0;
    
    for (int i = 0; i < apples_count; i++)
    {
        int sum = apples[i] + a;
        
        if (sum >= s && sum <= t)
        {
            landedApples++;
        }
    }
    
    for (int i = 0; i < oranges_count; i++)
    {
        int sum = oranges[i] + b;
        
        if (sum >= s && sum <= t)
        {
            landedOranges++;
        }
    }
    
    printf("%d\n%d", landedApples, landedOranges);
}

File: Practice/Problem Solving/Algorithms/Implementation/Apple and Orange.cs
using System;
using System.Linq;

class Solution
{
    void countApplesAndOranges(int s, int t, int a, int b, int[] apples, int[] oranges)
    {
        Console.WriteLine(apples.Count(x => x + a >= s && x + a <= t));
        
        Console.WriteLine(oranges.Count(x => x + b >= s && x + b <= t));
    }
}
